### PR TITLE
Pre-filter PII from Sentry events before they go over the wire

### PR DESCRIPTION
**Pull request by stevejalim**
As we've done with Bedrock and Basket, add pre-flight processing/PII filtering
to Nucleus.

Includes a simple test, ported from Basket, to confirm before_send() works as
we expect.

Also includes commented-out config to help future selves quickly add in
filtering or partial masking of sensitive vars

@robhudson After this is merged, can we chat about releasing it to prod - are there any non-standard steps you encountered?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -68,6 +68,9 @@ check-requirements: .make.docker.pull
 compile-requirements: .make.docker.pull
 	${DC} run --rm compile-requirements
 
+install-local-python-deps:
```
Comment (pmac) [inline Makefile:71]: did you mean to include this unrelated change or is it in need of a rebase?

**Review by pmac — APPROVED** [review]
🍰

File: Makefile (hunk 1 above)
Comment (stevejalim) [reply Makefile:71]: Ah, apols @pmac - it was a deliberate addition (matching the convenience commands added to Basket and Bedrock), but I should have put it in its own commit to make it clear it was deliberate

Comment (pmac) [reply Makefile:71]: Cool. Just making sure 👍
